Sync shootouts/submitted to CLBG site

bradcray opened the pull request:
This started as an intention to make sure our new binarytrees and
mandelbrot CLBG entries were in the shootouts/submitted directory
and then also ended up taking on the task of making sure that
our shootouts/submitted directory matches the CLBG site.  It ended
up growing a bit due to new changes to the regex redux benchmark.

More specifically:
* updated submitted binarytrees.chpl and added mandelbrot2.chpl as intended
* updated regex redux in release and submitted directory to update the
  new-new regex and updated the .perfkeys file accordingly to match
  what's reported on the site's runs
* updated input and output files for regex redux to match those suggested
  by the site
* added missing backslash to CLBG URL in pidigits code
* restore typo that made it onto the CLBG site (sorry @cassella)
* unify whitespace on fasta2.chpl (even though it's seriously uglier)

While here, I also looked at diffs with the release version and made these
improvements to them based on that:
* removed bogus trailing whitespace from release binarytrees that
  I hadn't noticed earlier, but which got submitted to the shootout site :'(
* put parens on release chameneos' deinit() function

bradcray commented:
@chapel-lang/perf-team : This is a thankless one, but can I get a review or at least blessing / sanity check on this?

psahabu commented on test/studies/shootout/submitted/spectralnorm.chpl:
```diff
@@ -38,7 +38,7 @@ proc multiplyAv(v: [?Dv], Av: [?DAv]) {
 
 //
-// Compute A-transpose * v ('Atv').
+// Compute A-tranpose * v ('Atv').
```
Shouldn't this just stay as "transpose"? Or are we being very stringent about the sync?

bradcray replied:
Stringent, to keep the diffs clean.  The version in the release (from which any future submissions will hopefully be derived) spells it correctly.

bradcray commented:
[reviewed by @psahabu]

This started as an intention to make sure our new binarytrees and
mandelbrot CLBG entries were in the shootouts/submitted directory
and then also ended up taking on the task of making sure that
our shootouts/submitted directory matches the CLBG site.  It ended
up growing a bit due to new changes to the regex redux benchmark.

More specifically:
* updated submitted binarytrees.chpl and added mandelbrot2.chpl as intended
* updated regex redux in release and submitted directory to update the
  new-new regex and updated the .perfkeys file accordingly to match
  what's reported on the site's runs
* updated input and output files for regex redux to match those suggested
  by the site
* added missing backslash to CLBG URL in pidigits code
* restore typo that made it onto the CLBG site (sorry @cassella)
* unify whitespace on fasta2.chpl (even though it's seriously uglier)

While here, I also looked at diffs with the release version and made these
improvements to them based on that:
* removed bogus trailing whitespace from release binarytrees that
  I hadn't noticed earlier, but which got submitted to the shootout site :'(
* put parens on release chameneos' deinit() function
